archey/entries/terminal.py

```diff
@@ -32,4 +85,37 @@ def __init__(self, *args, **kwargs):
             ) for i in range(37, 30, -1)
         ])
 
-        self.value = '{0} {1}'.format(terminal, colors)
+    @staticmethod
+    def _detect_terminal_emulator():
+        """Try to detect current terminal emulator based on various environment variables"""
+        # At first, try to honor `TERM_PROGRAM` environment variable.
+        # See <https://github.com/Maximus5/ConEmu/issues/1837#issuecomment-469199525>.
+        env_term_program = os.getenv('TERM_PROGRAM')
+        if env_term_program:
+            return env_term_program
+
+        # Second, check if we have any matches as defined in our `COLORTERM` constant dict.
+        env_colorterm = os.getenv('COLORTERM')
+        if env_colorterm:
+            for env_value_re, normalized_name in COLORTERM_DICT.items():
+                if re.match(env_value_re, env_colorterm):
+                    return normalized_name
+
+        # Third, check if we have any matches defined in our `TERM` constant dict.
+        env_term = os.getenv('TERM')
+        if env_term:
+            for env_value_re, normalized_name in TERM_DICT.items():
+                if re.match(env_value_re, env_term):
+                    return normalized_name
+            # If we didn't find any match and `TERM` is set to "something special", honor it.
+            if not env_term.startswith('xterm'):
+                return env_term
+
+        # If not, try to find a "known identifier" and perform name normalization...
+        for env_var, normalized_name in ENV_DICT.items():
+            if env_var in os.environ:
+                return normalized_name
+
+        # When nothing of the above matched, falls-back on the regular `TERM` environment variable.
+        # Note : It _might_ be `None` in very specific environments.
+        return env_term
```

Comment:
I've tried to keep very similar behaviour here while adding support for the `COLORTERM` environment variable (checking it before `TERM` but not using it unless an exact match is found)